// react-native-storybook-loader/out/logger/logger.js
"use strict";
exports.__esModule = true;
exports.BitBucketLogger = exports.ConsoleLogger = void 0;
var ConsoleLogger = /** @class */ (function () {
    function ConsoleLogger() {
    }
    ConsoleLogger.prototype.debug = function () {
        var data = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            data[_i] = arguments[_i];
        }
        console.debug.apply(console, data);
    };
    ConsoleLogger.prototype.error = function () {
        var data = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            data[_i] = arguments[_i];
        }
        console.error.apply(console, data);
    };
    ConsoleLogger.prototype.info = function () {
        var data = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            data[_i] = arguments[_i];
        }
        console.info.apply(console, data);
    };
    ConsoleLogger.prototype.log = function () {
        var data = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            data[_i] = arguments[_i];
        }
        console.log.apply(console, data);
    };
    ConsoleLogger.prototype.trace = function () {
        var data = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            data[_i] = arguments[_i];
        }
        console.trace.apply(console, data);
    };
    ConsoleLogger.prototype.warn = function () {
        var data = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            data[_i] = arguments[_i];
        }
        console.warn.apply(console, data);
    };
    return ConsoleLogger;
}());
exports.ConsoleLogger = ConsoleLogger;
var BitBucketLogger = /** @class */ (function () {
    function BitBucketLogger() {
    }
    BitBucketLogger.prototype.debug = function (_) { };
    BitBucketLogger.prototype.error = function (_) { };
    BitBucketLogger.prototype.info = function (_) { };
    BitBucketLogger.prototype.log = function (_) { };
    BitBucketLogger.prototype.trace = function (_) { };
    BitBucketLogger.prototype.warn = function (_) { };
    return BitBucketLogger;
}());
exports.BitBucketLogger = BitBucketLogger;
