// react-native-storybook-loader/out/rnstl-cli.js
#!/usr/bin/env node
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
exports.__esModule = true;
var yargs_1 = __importDefault(require("yargs"));
var fs_1 = require("fs");
var path_1 = __importDefault(require("path"));
var logger_1 = __importStar(require("./logger"));
var configuration_1 = require("./configuration");
var logger_2 = require("./logger");
var template_1 = require("./template");
var locator_1 = require("./locator");
var constants_1 = require("./constants");
var args = yargs_1["default"]
    .usage('$0 [options]')
    .options({
    searchDir: {
        type: 'string',
        array: true,
        desc: 'The directory or directories, relative to the project root, to search for files in.'
    },
    pattern: {
        desc: "Pattern to search the search directories with. Note: if pattern contains '**/*' it must be escaped with quotes",
        type: 'string'
    },
    outputFile: {
        desc: 'Path to the output file.',
        type: 'string'
    },
    silent: {
        desc: 'Silences all logging',
        type: 'boolean'
    }
})
    .help().argv;
logger_1.useConsoleLogger();
if (args.silent) {
    logger_1["default"].setLogLevel(logger_2.LogLevels.silent);
}
else {
    logger_1["default"].setLogLevel(logger_2.LogLevels.info);
}
logger_1["default"].debug('yargs', args);
(function () { return __awaiter(void 0, void 0, void 0, function () {
    var cwd, configuration, loaderDefinition, template, err_1;
    return __generator(this, function (_a) {
        switch (_a.label) {
            case 0:
                _a.trys.push([0, 6, , 7]);
                cwd = process.cwd();
                return [4 /*yield*/, configuration_1.generateConfiguration(args, cwd)];
            case 1:
                configuration = _a.sent();
                return [4 /*yield*/, locator_1.generateLoaderDefinition(configuration)];
            case 2:
                loaderDefinition = _a.sent();
                return [4 /*yield*/, template_1.generateTemplate(loaderDefinition)];
            case 3:
                template = _a.sent();
                return [4 /*yield*/, fs_1.promises.mkdir(path_1["default"].dirname(loaderDefinition.outputFile), {
                        recursive: true
                    })];
            case 4:
                _a.sent();
                logger_1["default"].info('Writing to ' + loaderDefinition.outputFile);
                return [4 /*yield*/, fs_1.promises.writeFile(loaderDefinition.outputFile, template, { encoding: constants_1.encoding })];
            case 5:
                _a.sent();
                return [3 /*break*/, 7];
            case 6:
                err_1 = _a.sent();
                logger_1["default"].error('Failed to execute: ' + err_1);
                return [3 /*break*/, 7];
            case 7: return [2 /*return*/];
        }
    });
}); })()["catch"](function (e) { return logger_1["default"].error(e); });
